lib/fluent/parser.rb

```diff
@@ -345,6 +352,16 @@ def convert_field_type!(record)
           end
         }
       end
+
+      def convert_value_to_nil(value)
+        if @null_empty_string
+          value = (value == '') ? nil : value
+        end
+        if @null_value_pattern
+          value = (value =~ @null_value_pattern) ? nil : value
```

Comment:
String scrub is needed? cf. http://blog.livedoor.jp/sonots/archives/34702351.html